pkg/kubelet/kubelet.go

```diff
@@ -133,6 +133,9 @@ const (
 	// Max amount of time to wait for the container runtime to come up.
 	maxWaitForContainerRuntime = 30 * time.Second
 
+	// Max amount of time to admit a single Pod, mostly accounting for device plugin allocation.
+	maxSinglePodAdmissionTimeout = 30 * time.Second
```

Comment:
this is the big question - is 30 seconds enough to admit a Pod? Will we break any existing customers with this?

Reply:
this could be a great case to guard a fix with a feature gate. There's at least another one for which I strongly feel we should add one. Linking the issue: https://github.com/kubernetes/kubernetes/issues/70585

In practice, wiring a context is little controversial. Once existing tests pass, we're reasonnably sure it's working, and I think such a PR can be merged quickly. Using it however, setting timeouts, possibly aborting admission with all the consequences to be checked, that should be done with care and possiblt guarded by a feature gate.